statsmodels/base/transform.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+from statsmodels.robust import mad
+from scipy.optimize import minimize_scalar
+
+
+class BoxCox(object):
+    """
+    Mixin class to allow for a Box-Cox transformation.
+    """
+
+    def transform_boxcox(self, x, lmbda=None, method='guerrero', **kwargs):
+        """
+        Performs a Box-Cox transformation on the data array x. If lmbda is None,
+        the indicated method is used to estimate a suitable lambda parameter.
+
+        Parameters
+        ----------
+        lmbda : float
+            The lambda parameter for the Box-Cox transform. If None, a value
+            will be estimated by means of the specified method.
+        method : {'guerrero', 'loglik'}
+            The method to estimate the lambda parameter. Will only be used if
+            lmbda is None, and defaults to 'guerrero', detailed in Guerrero
+            (1993). 'loglik' maximizes the profile likelihood.
+        **kwargs: dict
```

Comment:
**kwargs is not a dict, but are additional keyword arguments that the function accepts. These should be usual python variable names, so `r` and `scale`.